boto/ec2/elb/__init__.py

```diff
@@ -756,3 +757,71 @@ def detach_lb_from_subnets(self, name, subnets):
                                'Subnets.member.%d')
         return self.get_list('DetachLoadBalancerFromSubnets',
                              params, None)
+
+    # Tag methods
+
+    def _build_tag_param_list(self, params, tags):
+        keys = sorted(tags.keys())
+        i = 1
+        for key in keys:
+            value = tags[key]
+            params['Tags.member.%d.Key' % i] = key
+            if value is not None:
+                params['Tags.member.%d.Value' % i] = value
+            i += 1
+
+    def get_all_tags(self, load_balancer_names=None):
+        """
+        Retrieve all the metadata tags associated with your ELB(s).
+
+        :type load_balancer_names: list
+        :param load_balancer_names: An optional list of load balancer names.
+
+        :rtype: :class:`boto.ec2.elb.tag.TagDescriptions`
+        :return: A dictionary mapping load balancer names to
+            :class:`boto.ec2.elb.tag.TagSet`.
+        """
+        params = {}
+        if load_balancer_names:
+            self.build_list_params(params, load_balancer_names,
+                                   'LoadBalancerNames.member.%d')
+
+        return self.get_object('DescribeTags', params, TagDescriptions,
+                               verb='POST')
+
+    def add_tags(self, load_balancer_names, tags):
+        """
+        Create new metadata tags for the specified resource ids.
+
+        :type load_balancer_names: list
+        :param load_balancer_names: A list of load balancer names.
+
+        :type tags: dict
+        :param tags: A dictionary containing the name/value pairs.
+                     If you want to create only a tag name, the
+                     value for that tag should be the empty string
+                     (e.g. '').
+        """
+        params = {}
+        self.build_list_params(params, load_balancer_names,
+                               'LoadBalancerNames.member.%d')
+        self._build_tag_param_list(params, tags)
```

Comment:
with change above we can change it to

`params.update(self._build_tag_params(tags))`